astropy/modeling/utils.py

```diff
@@ -4,16 +4,31 @@
 This module provides utility functions for the models package.
 """
 
+from __future__ import annotations
+
 import warnings
 
 # pylint: disable=invalid-name
 from collections import UserDict
 from inspect import signature
+from typing import TYPE_CHECKING, overload
 
 import numpy as np
 
 from astropy import units as u
 
+if TYPE_CHECKING:
+    from collections.abc import Sequence
+    from typing import SupportsFloat, TypeVar
+
+    from numpy.typing import NDArray
+
+    from astropy.units import Quantity
+
+    DType = TypeVar("DType", bound=np.generic)
+    FloatLike = TypeVar("FloatLike", bound=SupportsFloat)
```

Comment:
Turns out we haven't used `FloatLike` anywhere since 4f2717101d3f6be1c007e68c42da7bdd5ef8b852.
```suggestion
```